pkg/sql/rowexec/joinreader.go

```diff
@@ -162,10 +201,23 @@ func newJoinReader(
 	}
 
 	var fetcher row.Fetcher
+	var colIdxMap map[sqlbase.ColumnID]int
+	var rightCols util.FastIntSet
+	switch readerType {
```

Comment:
Is it possible to not have to do this branch many different spots? It's fine if not.